### DISCUSSION TITLE
DotNet pack triggers MINVER1001 for Core 3.1

### DISCUSSION THREAD

**Discussion by JimBobSquarePants** [adamralph/minver#518] [Q&A] [answered]
This is a very odd issue and I cannot figure out the cause. I thought I'd better open a discussion rather than an issue since it's likely something I've done.

Basically for Core 3.1 only in a multitargeted project MinVer is failing to find the correct version.

>MinVer : warning MINVER1001: 'D:\a\ImageSharp.Drawing\ImageSharp.Drawing\src\ImageSharp.Drawing' is not a valid working directory. Using default version 0.0.0-alpha.0. [D:\a\ImageSharp.Drawing\ImageSharp.Drawing\src\ImageSharp.Drawing\ImageSharp.Drawing.csproj]
  MinVer: Calculated version 0.0.0-alpha.0.
  
 Inspecting the dlls shows us the correct version is applied for everything except .NET Core 3.1
 
.NET Core 2.1 👍  

![image](https://user-images.githubusercontent.com/385879/119532051-70250200-bd7c-11eb-8b4e-80ded9414067.png)

 .NET Core 3.1 👎 
 
![image](https://user-images.githubusercontent.com/385879/119532228-9b0f5600-bd7c-11eb-82d2-53faaafbbf1b.png)

 We've been using the same setup across all our repositories but this one is the only one to have issues. 
 
 https://github.com/SixLabors/ImageSharp.Drawing/issues/137

**Comment by adamralph** [comment, marked as answer]
@JimBobSquarePants can you reproduce this locally or is it just in CI?

**Reply by JimBobSquarePants** [reply]
I haven’t been able to reproduce it locally as yet. Still trying.

**Reply by JimBobSquarePants** [reply]
Still unable to recreate locally but I have determined that the issue is Windows only. Am able to use Ubuntu as a workaround.

**Reply by bording** [reply]
It would be helpful if you could capture a binlog of the failed CI run. That would make it easier to understand what is going on.

You'd need to share it privately though if you have any secrets stored in environment variables since those would be included in the log file.

**Reply by JimBobSquarePants** [reply]
Will see what I can do.

**Reply by adamralph** [reply]
@JimBobSquarePants it would also be good to set this in your csproj file so that we can see what Git is doing when invoked by MinVer:

```xml
<PropertyGroup>
  <MinVerVerbosity>diagnostic</MinVerVerbosity>
</PropertyGroup>
```

**Reply by adamralph** [reply]
@JimBobSquarePants are you still having this problem? If so, can you please provide the diagnostics we requested above?

**Reply by JimBobSquarePants** [reply]
Sorry, haven't had the change to debug. I worked around by changing OS to Ubuntu but I've kept the branch with the issue to test against so if you can give me a few days to get it tested before closing that would be ace.
